### DISCUSSION TITLE
v 1.14.1

### DISCUSSION THREAD

**Discussion by spotlessmind1975** [spotlessmind1975/ugbasic#627] [Announcements]
### **New features**

   * Added support for IMPORT DECLARES.
   * Added support for SYSCALL.
   * Added support for OUT(put) parameters on DECLARE.
   * Added support for DECLARE SYSTEM FUNCTION and calling order.
   * COCO specific cursor #620
   * Added support for DECLARE PROC / FUNCTION.
   * Allow inclusion of ASM snippets #156
   * Add support for SYS command #300

<hr /><em>This discussion was created from the release <a href='https://github.com/spotlessmind1975/ugbasic/releases/tag/v1.14.1'>v 1.14.1</a>.</em>

**Comment by pwillard** [comment]
KEYBOARD DELAY AND KEYBOARD RATE seem to help a lot... but the Help description seems backward.  It works properly (for me) with the INPUT command when RATE and DELAY = Low Numbers (like 1) .  It is a relief to see INPUT working without repeated characters.  With 254 value in for DELAY and RATE, I get --->  
![254DELET254RATE](https://github.com/spotlessmind1975/ugbasic/assets/3194144/a91473de-6c65-478a-ad4e-1522bc20ae50)

**Reply by spotlessmind1975** [reply]
Hello @pwillard , and **thank you for your comment!**

The command `DEFINE KEYBOARD RATE` define a sort of "typematic counter" for the repetition of each key pressed. So a low number means an high counter, and viceversa. Actually. by experimental tries done using XRoar, it means:

      ' Set a speed of about 5 cps (characters per second)
      DEFINE KEYBOARD RATE 1

      ' Set a speed of about 25 cps (characters per second)
      DEFINE KEYBOARD RATE 100

      ' Set a speed of about 50 cps (characters per second)
      DEFINE KEYBOARD RATE 200

And so on. Obviously, if you put `254` it means more than 70-80 cps, that is something similar to a "no delay" between repetitions. Currently, the command `DEFINE KEYBOARD DELAY` is not available under TRS-80 Color Computer. 

Thank you again and have nice retrocoding!
